app/javascript/react/pages/ApplicantsUpload.jsx

```diff
@@ -141,14 +142,25 @@ export default function ApplicantsUpload({ organisation, configuration, departme
   };
 
   const retrieveUpToDateApplicants = async (applicantsFromList) => {
+    const departmentInternalIds = applicantsFromList
+      .map((applicant) => applicant.departmentInternalId)
+      .filter((departmentInternalId) => departmentInternalId);
     const uids = applicantsFromList.map((applicant) => applicant.uid).filter((uid) => uid);
-    let upToDateApplicants = applicantsFromList;
 
-    const retrievedApplicants = await retrieveApplicantsFromApp(uids);
+    const retrievedApplicants = await retrieveApplicantsFromApp(departmentInternalIds, uids);
+
+    const upToDateApplicants = applicantsFromList.map((applicant) => {
+      const upToDateApplicant = retrievedApplicants.find((a) =>(a.department_internal_id && a.department_internal_id === applicant.departmentInternalId) || (a.uid && a.uid === applicant.uid));
 
-    upToDateApplicants = applicantsFromList.map((applicant) => {
-      const upToDateApplicant = retrievedApplicants.find((a) => a.uid === applicant.uid);
       if (upToDateApplicant) {
+        if (applicant.departmentInternalId && !upToDateApplicant.department_internal_id) {
+          upToDateApplicant.department_internal_id = applicant.departmentInternalId
+          updateApplicant(
+            upToDateApplicant.department_id,
+            upToDateApplicant.id,
+            upToDateApplicant
+          );
+        }
```

Comment:
Je comprends l'idée, mais comme je t'ai dit je suis pas fan de rajouter de la complexité "cachée". Surtout qu'on sait jamais ce qui peut se passer en cas d'erreur. 
Moi je pense qu'on pourrait proposer une action upddate pour tous les attributs s'ils sont différents de ce qu'il y a en base (voir #201), et dans ce cas-là enlever cette update automatique. Ça permettrait de rendre visibles les cas d'erreurs qu'on aurait pas anticiper.

Reply:
Je comprends pas comment tu voudrais faire cette action update. Proposer à l'utilisateur s'il veut updater à chaque fois ?
C'est à dire demander x fois à l'agent de regarder la comparaison et lui faire valider à chaque fois ? Ca me paraît un enfer du point de vue utilisateur. Effectivement, il faudrait peut-être prévoir une manière de gérer les éventuelles erreurs (qui devraient être extrêmement rares je pense), mais je trouve pas que ce soit spécialement plus de la "complexité cachée" que plein d'autres choses que l'on fait déjà under the hood.

Reply:
@qblanc de quelles autre `write` actions parles-tu ? Je ne veux pas "proposer" directement, on aurait dans ces cas-là un bouton "Mettre à jour" au lieu de "Créer compte".

Reply:
Autant pour les données de contact il y a des arguments pour ne faire d'update automatique, avec lesquels je ne suis pas forcément d'accord, mais qui sont tout à fait légitimes, autant là je ne vois pas le moindre inconvénient à faire ça (à part une opposition théorique/de principe à faire des updates "cachées").
1) Si on ne le fait pas, que l'on charge un utilisateur déjà créé qui a un ID éditeur dans le fichier mais qui n'en a pas en base, alors il sera updaté avec `upToDateApplicant`, ce qui aura pour conséquence de faire sauter l'information sur l'id éditeur, purement et simplement.
2) Si on le fait et que l'update se passe bien (ce qui devrait se passer dans 99% des cas), alors le système peut récupérer un ID éditeur pour les utilisateurs qui auraient été créés sans.
3) Si on le fait et que ça se passe mal, il se passe ce qu'il se passerait à l'étape 1) ; donc, il n'y a aucun inconvénient à le faire.

Du point de vue de l'agent, tout cela n'a de toute façon aucune importance, il se moque un peu de connaître ou pas l'ID éditeur du bénéificaire : c'est uniquement pour nous aider à mieux retrouver les bRSA/éviter les doublons ; c'est autant un update "caché" que quand on récupère l'id rdv-s pour synchroniser les deux profils, à mon sens. Il n'y a aucune raison de lui "infliger" un clic supplémentaire pour quelque chose dont il se moque, qui ne fera que le perturber et alors même qu'il n'y a, de mon point de vue, aucun cas de figure où celà vaut le coup de ne pas déclencher l'action.

Reply:
@qblanc je vois pas pourquoi tu parles d'opposition théorique ou de principe. 
Envoyer des requêtes à la volée qui écrivent en base au moment de l'upload pourraient avoir des conséquences néfastes pour l'application sans qu'on le remarque. Même si les risques sont faibles (et j'en suis pas certain) je préfère pas prendre le risque. Sans compter le fait que niveau load ça pourrait envoyer +1000 requêtes d'un coup à notre appli (et il me semble qu'il y avait eu un problème du type entre analyse flux inserton et RDV-S).

Reply:
Je comprends ton point de vue et je m'y suis rangé, mais je n'ai pas la même évaluation des risques on va dire. Pour les + 1000 requêtes les chances que ça arrive sont extrêmement faibles  : les requêtes ne seraient envoyées que pour les utilisateurs qui sont déjà créés (ce qui n'est pas fréquent sur cette page normalement), et quand ils leur manquent des contacts. Même si il y avait des fichiers stocks de 2000 demandeurs, comme pour les Ardennes, il grossit au fur et à mesure, avec des utilisateurs créés au fur et à mesure, et qui se verraient donc assigner au fur et à mesure leurs contacts ; une fois que c'est fait, ils ne reçoivent plus de requêtes. Et quand bien même, dans le pire des cas, on constate un problème de charge, ce n'est pas dramatique, il serait facile d'enlever/de corriger ce qui pose souci. Mais whatever, ça me va très bien comme ça.

Car en vrai, il y avait un souci auquel je n'avais pas pensé initialement, ce serait les cas où : un utilisateur a été créé disons il y a 1 mois, sans mail mais avec un téléphone. L'invitation mail a été envoyée. Un mois plus tard, le fichier contact lui trouve un téléphone (dans le cas où le fichier upload est un stock, comme pour les Ardennes). Cela va débloquer le bouton "Invitation SMS", et si l'agent clique un peu en mode bourrin sur tous les boutons débloqués (ce qui à mon avis doit arriver avec certains agents), il va se retrouver à envoyer une invitation à quelqu'un qui en avait reçu une un mois avant et a peut-être déjà eu son rdv. Perso c'est principalement cette raison qui m'a fait me ranger à ton avis :-)

Tu vois d'autres modifs à faire ?

Reply:
Non je pense que c'est bon après ça 👍